### ISSUE TITLE
ablate_head masks MultiheadAttention output after head mixing

### ISSUE THREAD

**Issue by Sagar300100** [l3hargav/Grokking#1]
`ablate_attention.py` L68–86 hooks `layer.self_attn` (an `nn.MultiheadAttention` inside `nn.TransformerEncoderLayer`) and zeroes `attn_out[:, :, start:end]`. `nn.MultiheadAttention` applies `out_proj` before returning, so that slice is `HEAD_DIM` output coordinates that every head wrote to, not head `head_idx`. Both `ablate_head` and `ablate_multiple_heads` are affected.

Fix: a pre-hook on `out_proj` may never fire, because `nn.MultiheadAttention` passes `out_proj.weight` and `out_proj.bias` straight into `F.multi_head_attention_forward` rather than calling the module. Zero the head's input columns of the projection weight on a copy instead:

```python
import copy
ablated = copy.deepcopy(model)
W = ablated.transformer.layers[0].self_attn.out_proj.weight   # [d_model, n_heads*HEAD_DIM]
with torch.no_grad():
    W[:, head_idx*HEAD_DIM:(head_idx+1)*HEAD_DIM] = 0
# evaluate `ablated`; discard it afterwards
```

Scope: source inspection only; I haven't rerun the lesion experiments and am not claiming how the reported effects change.
